src/OrchardCore/OrchardCore.Testing/ModuleNamesProvider.cs

```diff
@@ -0,0 +1,24 @@
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using OrchardCore.Modules;
+using OrchardCore.Modules.Manifest;
+
+namespace OrchardCore.Testing
+{
+    public class ModuleNamesProvider : IModuleNamesProvider
+    {
+        private readonly IEnumerable<string> _moduleNames;
+
+        public ModuleNamesProvider()
```

Comment:
public ModuleNamesProvider(Assembly asm)
pass the Assembly you want to use for Module Discovery